### DISCUSSION TITLE
Question: Is possible to relay automatically, the stream to youtube/twich/facebook/other rtsp server?

### DISCUSSION THREAD

**Discussion by maxlinux2000** [bluenviron/mediamtx#270] [Questions]
## Which version are you using?

rtsp-simple-server_v0.9.3_linux_arm64.tar.gz

## Describe the problem

I need to know if it is possible to relay the current stream to other rtsp servers

Imagine that I am using rtsp-simple-server on my server with obs but my line cannot support more than 100 connections at a time, and I want to send users on youtube for example (sent the link in my telegram group for example) and thus reserving my line only for some local users, such as local cable tv.

Can it be done directly from rtps-simple-server? how?

regards
MaX

**Comment by maxlinux2000** [comment]
mmmm, maybe this option is something like what I'm looking for

     # command to run when a client starts publishing.
     # This is terminated with SIGINT when a client stops publishing.
     runOnPublish:

if I run a script in runOnPublish that takes the stream and re-sends it to another rtsp server, will I get what I'm looking for, will you?

something like:

```
#!/bin/bash
# Server URL
Avideo_URL=rtmp://live-mad.twitch.tv/app/
# Stream name/key
Key=live_123123123_fjeiworfmkflsdkflsdmfwerioweor
SOURCE="rtsp://localhost:8554/mystream" 
ffmpeg \
-i "$SOURCE" \
-c:a copy \
-bsf:a aac_adtstoasc \
-c:v copy \
-f flv "$Avideo_URL/$Key"

```

edit... no the script don't run :-(

**Comment by aler9** [comment]
Hello, yes, you should get the result by using runOnPublish.
Unfortunately most streaming providers support only RTMP, which is a non-standardized protocol that supports only TCP (less efficient but more supported), so the script has to remux the stream.

[a comment by an automated account is not shown]
